Description: Add possibility to operate a single worker from init script
Author: Brice Goglin <bgoglin@debian.org>
Reviewed-by: Andriy Senkovych <jolly_roger@itblog.org.ua>
Origin: other, https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=661102
Bug-Debian: http://bugs.debian.org/661102
Forwarded: https://github.com/buildbot/buildbot/pull/1267
Date: Fri, 24 Feb 2012 09:11:27 +0100
Last-Update: 2014-10-19

Index: debian-buildbot-worker/contrib/init-scripts/buildbot-worker.init.sh
===================================================================
--- debian-buildbot-worker.orig/contrib/init-scripts/buildbot-worker.init.sh
+++ debian-buildbot-worker/contrib/init-scripts/buildbot-worker.init.sh
@@ -148,7 +148,9 @@ function worker_op () {
 function do_op () {
     errors=0
     for i in $( seq ${#WORKER_ENABLED[@]} ); do
-        if is_disabled "${WORKER_ENABLED[$i]}" ; then
+        if [ -n "$4" ] && [ "$4" != "${WORKER_NAME[$i]}" ] ; then
+            continue
+        elif is_disabled "${WORKER_ENABLED[$i]}" && [ -z "$4" ] ; then
             continue
         fi
 
@@ -175,19 +177,19 @@ function do_op () {
 
 case "$1" in
     start)
-        do_op "worker_op" "start" "Starting buildbot worker"
+        do_op "worker_op" "start" "Starting buildbot worker" "$2"
         exit $?
         ;;
     stop)
-        do_op "worker_op" "stop" "Stopping buildbot worker"
+        do_op "worker_op" "stop" "Stopping buildbot worker" "$2"
         exit $?
         ;;
     reload)
-        do_op "worker_op" "reload" "Reloading buildbot worker"
+        do_op "worker_op" "reload" "Reloading buildbot worker" "$2"
         exit $?
         ;;
     restart|force-reload)
-        do_op "worker_op" "restart" "Restarting buildbot worker"
+        do_op "worker_op" "restart" "Restarting buildbot worker" "$2"
         exit $?
         ;;
     *)
